### PR TITLE
Fix pos_label not propagated to evaluate ROC/PR/calibration curves

### LINKED ISSUE #23991: [BUG] mlflow.evaluate: precision_recall_auc and ROC/PR/calibration curves are wrong for a non-default pos_label
> [!WARNING]
> Before submitting a PR, please make sure that:
> - A maintainer has triaged this issue and applied the `ready` label
> - This issue has no assignee
> - No duplicate PR exists
>
> PRs not meeting these requirements may be automatically closed.

### Issues Policy acknowledgement
- [x] I have read and agree to submit bug reports in accordance with the [issues policy](https://www.github.com/mlflow/mlflow/blob/master/ISSUE_POLICY.md)

### Where did you encounter this bug?
Local machine

### MLflow version
- Client: 3.12 (current master)

### Describe the problem

When `pos_label` is set to the class that **isn't** the higher-sorted label, `mlflow.evaluate(model_type="classifier")` reports an incorrect `precision_recall_auc` and produces ROC / precision-recall / calibration **curve artifacts for the wrong class**.

The scalar `roc_auc` looks correct because ROC AUC is symmetric under a positive/negative class swap, which masks the problem — but `precision_recall_auc` is not symmetric and comes out wrong, and the logged curve plots are computed against the wrong probability column.

This is the common imbalanced-binary case where the minority/positive class is the lower-sorted label (fraud, churn, default-risk).

### Code to reproduce issue

```python
import numpy as np, pandas as pd, mlflow, sklearn.linear_model
from sklearn.metrics import average_precision_score

mlflow.set_tracking_uri("sqlite:///mlruns.db")
rng = np.random.RandomState(0)
X = rng.randn(400, 4)
y = (X.sum(1) > 0).astype(int)            # well-separated binary problem, classes {0, 1}
model = sklearn.linear_model.LogisticRegression().fit(X, y)
P = model.predict_proba(X)                # columns follow model.classes_ = [0, 1]

# Correct average-precision for positive class 0, using P(class 0):
print("correct pr_auc, pos_label=0:", round(average_precision_score(y == 0, P[:, 0]), 4))  # 0.9998

data = pd.DataFrame(X, columns=[f"f{i}" for i in range(4)]); data["label"] = y
with mlflow.start_run():
    mi = mlflow.sklearn.log_model(model, name="m")

for pl in (1, 0):
    with mlflow.start_run():
        res = mlflow.evaluate(mi.model_uri, data, targets="label", model_type="classifier",
                              evaluator_config={"pos_label": pl, "label_list": [0, 1]})
        print(f"mlflow pos_label={pl}: roc_auc={res.metrics['roc_auc']:.4f} "
              f"pr_auc={res.metrics['precision_recall_auc']:.4f}")
```

### Observed

```
correct pr_auc, pos_label=0: 0.9998
mlflow pos_label=1: roc_auc=0.9997 pr_auc=0.9997
mlflow pos_label=0: roc_auc=0.9997 pr_auc=0.3237   <-- wrong (should be ~0.9998)
```

`pos_label=1` (the higher-sorted, default-style positive class) is correct; only the non-default positive class is affected.

### What component(s) does this bug affect?
- [x] `area/evaluation`: MLflow Evaluation APIs, `mlflow.evaluate`

### Additional context

I've traced the root cause and have a fix with regression tests ready locally. Happy to open a PR — could a maintainer triage and apply the `ready` label so the PR isn't auto-closed?

### PR DESCRIPTION
**Pull request by immu4989**
### Related Issues/PRs

Closes #23991

### What changes are proposed in this pull request?

`mlflow.evaluate(model_type="classifier")` computed the binary `precision_recall_auc` and the ROC / precision-recall / calibration **curve artifacts** against a hardcoded probability column (`y_probs[:, 1]`). `_validate_label_list` reorders `label_list` so `pos_label` is last, but the predicted-probability columns stay in the model's (sorted) class order. As a result, when `pos_label` is **not** the higher-sorted class:

- `precision_recall_auc` is wrong (e.g. `0.32` instead of `~0.99` on a separable problem),
- the ROC / precision-recall / calibration curve plots are produced for the wrong class.

The scalar `roc_auc` happened to look correct because ROC AUC is symmetric under a positive/negative swap, which masked the bug.

This PR:
- captures the probability column matching `pos_label` (before `label_list` is reordered) and uses it when generating the binary ROC/PR curves in `classifier.py`,
- selects the `pos_label` column in `plot_calibration_curve` instead of assuming column index 1,
- makes the ROC AUC scalar respect `pos_label` by binarizing `y_true` against it.

A targeted, localized fix was chosen over reordering `y_probs` globally because `plot_lift_curve` derives its own class order from `np.unique(y_true)`; a global reorder would have silently broken the lift curve.

### How is this PR tested?

- [x] New unit/integration tests

Added `test_evaluation_binary_classification_curve_auc_respects_pos_label`, which asserts `roc_auc` and `precision_recall_auc` from `mlflow.evaluate` match scikit-learn's values for the configured positive class. It fails on `master` for the non-default `pos_label` and passes with this change; the default `pos_label` is unaffected.

### Does this PR require documentation update?

- [x] No.

### Does this PR require updating the [MLflow Skills](https://github.com/mlflow/skills) repository?

- [x] No.

#### Is this a user-facing change?

- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

Fixed a bug where `mlflow.evaluate` reported an incorrect `precision_recall_auc` and produced ROC / precision-recall / calibration curves for the wrong class when `pos_label` was set to the non-default class in binary classification.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [x] `area/evaluation`: MLflow model evaluation features, evaluation metrics, and evaluation workflows

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes

#### Is this PR a critical bugfix or security fix that should go into the next patch release?

<details>
<summary>What is a minor/patch release?</summary>

- Minor release: a release that increments the second part of the version number (e.g., 1.2.0 -> 1.3.0).
  Minor releases are expected to contain larger changes, such as new features and improvements. Non-critical bug fixes and doc updates can be included as well. By default, your PR should target the next minor release.
- Patch release: a release that increments the third part of the version number (e.g., 1.2.0 -> 1.2.1).
  Patch releases are typically only performed when there has been a major regression or bug in the latest release. For the sake of stability, your PR should not be included in a patch release unless it is a critical fix, or if the risk level of your PR is exceedingly low.

</details>

- [ ] This PR is critical and needs to be in the next patch release
- [x] This PR can wait for the next minor release

### CODE REVIEW

**Review by B-Step62 — APPROVED** [review]
LGTM
